pkg/tests/combined_account_transition_test.go

```diff
@@ -0,0 +1,269 @@
+package tests
+
+import (
+	"testing"
+
+	"github.com/iotaledger/hive.go/log"
+	"github.com/iotaledger/iota-core/pkg/model"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/accounts"
+	"github.com/iotaledger/iota-core/pkg/testsuite"
+	"github.com/iotaledger/iota-core/pkg/testsuite/mock"
+	iotago "github.com/iotaledger/iota.go/v4"
+	"github.com/iotaledger/iota.go/v4/tpkg"
+)
+
+// Test_AccountStateTransition follows the account state transition flow described in:
+// https://github.com/iotaledger/iota-core/issues/660#issuecomment-1892596243
+func Test_AccountStateTransition(t *testing.T) {
+	ts := testsuite.NewTestSuite(t,
+		testsuite.WithProtocolParametersOptions(
+			iotago.WithTimeProviderOptions(
+				0,
+				testsuite.GenesisTimeWithOffsetBySlots(200, testsuite.DefaultSlotDurationInSeconds),
+				testsuite.DefaultSlotDurationInSeconds,
+				testsuite.DefaultSlotsPerEpochExponent,
+			),
+		),
+	)
+	defer ts.Shutdown()
+
+	node1 := ts.AddValidatorNode("node1")
+	node2 := ts.AddValidatorNode("node2")
+	wallet := ts.AddDefaultWallet(node1)
+
+	ts.Run(true)
+
+	node1.Protocol.SetLogLevel(log.LevelTrace)
+
+	// split genesis output into 4 outputs for the further usage.
+	tx1 := wallet.CreateBasicOutputsEquallyFromInput("TX1", 4, "Genesis:0")
+	ts.IssueBasicBlockWithOptions("block0", wallet, tx1)
+
+	// Issue some more blocks to make transaction accepted
+	{
+		ts.IssueValidationBlockWithHeaderOptions("vblock0", node2, mock.WithStrongParents(ts.BlockID("block0")))
+		ts.IssueValidationBlockWithHeaderOptions("vblock1", node1, mock.WithStrongParents(ts.BlockID("vblock0")))
+		ts.IssueValidationBlockWithHeaderOptions("vblock2", node2, mock.WithStrongParents(ts.BlockID("vblock1")))
+
+		ts.AssertTransactionsInCacheAccepted(wallet.Transactions("TX1"), true, node1, node2)
+	}
+
+	// create the account1 from TX1:0 with wallet "first"
+	// generated (block1, TX2)
+	ts.AddWallet("first", node1, iotago.EmptyAccountID)
+	createFullAccount(ts)
+
+	// create the account2, from implicit to full account from TX1:1 with wallet "second"
+	// generated (block2, TX3), (block3, TX4)
+	ts.AddWallet("second", node1, iotago.EmptyAccountID)
+	implicitAccID := createImplicitToFullAccount(ts)
```

Comment:
At this point, it is already a "normal account ID".

Reply:
It's a normal account, will rename it.